### PR TITLE
Standardise GitHub downloads and updates

### PR DESCRIPTION
**Pull request by tonytheodore**

### CODE REVIEW

**Comment by tonytheodore** [comment]
I think this ready to go now.

File: Makefile [hunk 1]
```diff
@@ -467,6 +472,9 @@ endef
 $(foreach TARGET,$(MXE_TARGETS),$(call TARGET_RULE,$(TARGET)))
 
 define PKG_RULE
+# configure GitHub metadata if GH_CONF is set
+$(and $($(PKG)_GH_CONF),$(eval $(MXE_SETUP_GITHUB)))
```
Comment (starius) [inline Makefile]: Can you use `$(if` here instead of `$(and`?

Comment (tonytheodore) [inline Makefile]: [Done](https://github.com/mxe/mxe/pull/1712/commits/4ed90733c937c8549fc926165781c025c380dec2)

**Comment by tonytheodore** [comment]
Filtering [clarified and improved using `sed`](https://github.com/tonytheodore/mxe/compare/4ed9073...tonytheodore:3dcad42)

**Comment by tonytheodore** [comment]
Messages in `update-package` [are now suppressed](https://github.com/mxe/mxe/pull/1712/commits/b2d99407c0d59f5149c9791ef1202abf7147e069):

```sh
git diff --name-only master \
    | sed -n 's,src/\(.*\)\.mk,\1,p' \
    | xargs -I pkg make check-update-package-pkg
NEW      libevent  2.0.21 --> 2.1.8
.        libffi  3.2.1
.        libsoup  2.57.1
NEW      qtserialport_qt4  5c3b6cc --> 56ebaec
NEW      vigra  1.9.0 --> 1.11.0
NEW      vmime  3cfcdca --> 1bc9f45
```

**Comment by starius** [comment]
Looks good to me!

**Comment by tonytheodore** [comment]
Okay, I'll merge it and see if I can find more edge cases.

**Pull request merged**